packages/adapter/lib/adapter/adapter.js

```diff
@@ -6723,112 +6717,9 @@ function Adapter(options) {
          *        See possible attributes of the state in @setState explanation
          */
         this.getState = (id, options, callback) => {
-            if (typeof options === 'function') {
-                callback = options;
-                options = {};
-            }
-
-            if (!adapterStates) { // if states is no longer existing, we do not need to unsubscribe
-                this.log.info('getState not processed because States database not connected');
-                return tools.maybeCallbackWithError(callback, tools.ERRORS.ERROR_DB_CLOSED);
-            }
-
-            if (!adapterObjects) {
-                this.log.info('getState not processed because Objects database not connected');
-                return tools.maybeCallbackWithError(callback, tools.ERRORS.ERROR_DB_CLOSED);
-            }
-
-            try {
-                validateId(id, false, null);
-            } catch (err) {
-                return tools.maybeCallbackWithError(callback, err);
-            }
-
+            // get state does the same as getForeignState but fixes the id first
             id = this._fixId(id, false);
-
-            if (options && options.user && options.user !== SYSTEM_ADMIN_USER) {
-                checkStates(id, options, 'getState', err => {
```

Comment:
on `getState` this was suboptimal, because the object can be used from the callback here, instead a few lines later we have called `getObject` again. In `getForeignState` this has been optimized.. just one downside of duplicate code